core/src/main/java/com/google/zxing/datamatrix/decoder/DecodedBitStreamParser.java

```diff
@@ -555,4 +588,70 @@ private static int unrandomize255State(int randomizedBase256Codeword,
     return tempVariable >= 0 ? tempVariable : tempVariable + 256;
   }
 
+  private static final class ECIStringBuilder {
```

Comment:
Let's refactor this into a package-private class in the .common package and reuse it across this and QR codes

Reply:
Just to be sure, in this PR I did not modify decoder code like this class. The commits appear in this PR but they are really two weeks old (I am still struggling with GitHub and occasionally this happens that it is shown like this).
It you are OK with it, then I can look in a separate PR if the ECIStringBuilder class has common code with the QR-Code decoder. I did however now move some code that is common to the two encoders into the common package.